Sources/PT.PM.Matching/Patterns/PatternStatements.cs

```diff
@@ -46,7 +46,12 @@ public override MatchingContext Match(Ust ust, MatchingContext context)
 
             if (Statements == null || Statements.Count == 0)
             {
-                return context;
+                if(blockStatement.Statements == null || blockStatement.Statements.Count == 0)
+                {
+                    context = context.AddMatch(blockStatement);
+                    return context;
```

Comment:
Do not use early return for success matching. Success matches should be returned only in a single exit point (see other samples). Failed matches can be returned anywhere.